pages/tech/data-science/style.md

```diff
@@ -12,47 +13,86 @@ We mostly follow the standard Python style conventions from [PEP8](https://www.p
 
 We use a series of code linters to maintain consistent formatting across our projects. Most projects will also use `pre-commit` to automate regular execution of the linters. The linters are also regularly run as part of our continuous integration and [testing](testing) suite.
 
-### Black
+### Ruff
 
-[Black](https://black.readthedocs.io/en/stable/index.html) is an opinionated PEP-compliant code formatter. We use Black's default settings with a few minor adjustments:
+[Ruff](https://docs.astral.sh/ruff) is a Python linter and code formatter which supports a wide range of linting rules, many of which are derived from the popular tools like [Flake8](https://flake8.pycqa.org/en/latest/) and [isort](https://pycqa.github.io/isort/), [pyupgrade](https://github.com/asottile/pyupgrade) and others. Ruff provides a formatter designed to be used as a drop-in replacement for [Black](https://black.readthedocs.io/en/stable/index.html) -- an opinionated PEP-compliant code formatter.
 
-Example `pyproject.toml`:
-
-```ini
-[tool.black]
-line-length = 100
-skip-string-normalization = true
-```
-
-### Flake8
-
-[Flake8](https://flake8.pycqa.org/en/latest/) provides additional code formatting sytle checks.
-
-Example `setup.cfg`:
-
-```ini
-[flake8]
-ignore = E203,E266,E501,W503,E722,E402,C901
-max-line-length = 100
-max-complexity = 18
-select = B,C,E,F,W,T4,B9
-```
+We use Ruff's default settings with a few minor adjustments:
```

Comment:
Do we comply with this internally in practice and should we use something other then the default?

Reply:
Also, I could see the .md definition of the `.pre-commit-config.yaml` going quickly out of date.